src/registry.rs

```diff
@@ -83,47 +124,47 @@ impl Registry {
         self.r.write().unwrap().unregister(c)
     }
 
-    pub fn write_pb<T: Write>(&self, _: &mut T) -> Result<()> {
-        Ok(())
+    fn gather(&self) -> Vec<proto::MetricFamily> {
+        self.r.read().unwrap().gather()
     }
 
-    pub fn write_text<T: Write>(&self, _: &mut T) -> Result<()> {
-        Ok(())
-    }
-}
+    pub fn scrap(&self, writer: &mut Write, encoder: &Encoder) -> Result<usize> {
```

Comment:
It is not clean for outer use, do we write first and then encode or vice versa? 
As I said before, we can only support gather + encoder and let user decide how to use outer, but if you persist on using this function, please think a better name and add comment for the arguments.